src/miv_simulator/mechanisms/__init__.py

```diff
@@ -59,10 +60,20 @@ def compile(directory: Optional[str] = None, force: bool = False) -> str:
 def compile_and_load(
     directory: Optional[str] = None, force: bool = False
 ) -> str:
-    """Compile and load dll file into NEURON"""
-    src = compile(directory, force)
+    """
+    Compile mechanism file on the processor 0, and load the output DLL file into NEURON.
+    """
+    comm = MPI.COMM_WORLD
+    rank = comm.rank
+    if rank == 0:
+        src = compile(directory, force)
+    else:
+        src = None
+    comm.barrier()
+    src = comm.bcast(src, root=0)
+
```

Comment:
I think this logic shouldn't be part of the method, since people may like to use the method without MPI. How about adding a decorator utility? Something like:

```python
def mpi_once(f: Callable) -> Any:
     def _wrap(*args, **kwargs):
        comm = MPI.COMM_WORLD
        rank = comm.rank
        result = None
        if rank == 0:
            result = f(*args, **kwargs)
        comm.barrier()
        result= comm.bcast(result, root=0)
        return result
     return _wrap
```
That would make this a bit more modular and a generic utility
```python
src = mpi_once(compile_and_load)(force=True)
```

Reply:
@frthjf I thought the code can still be executed without MPI, and `comm.rank` will be set to 0. Do you have any machine for a quick test?

I agree that having this utility could clean up the code in many places.

Reply:
Good point. I haven't tested it but yeah [looks like it's just 0](https://github.com/mpi4py/mpi4py/blob/master/src/mpi4py/MPI/Comm.pyx#L3256) so should be good actually.